libs/payments/ui/src/lib/client/components/CouponForm/index.tsx

```diff
@@ -88,43 +90,48 @@ const WithoutCoupon = ({
   const [error, formAction] = useFormState(applyCoupon, null);
 
   return (
-    <div className="bg-white rounded-b-lg shadow-sm shadow-grey-300 mt-6 p-4 rounded-t-lg text-base tablet:my-8">
-      <h2 className="m-0 mb-4 font-semibold text-grey-600">
-        <Localized id="next-coupon-promo-code">Promo Code</Localized>
-      </h2>
+    <Form.Root
+      action={formAction}
+      className="bg-white rounded-b-lg shadow-sm shadow-grey-300 mt-6 p-4 rounded-t-lg text-base tablet:my-8"
+      data-testid="coupon-form"
+    >
+      <Form.Field name="couponCode">
+        <Localized id="next-coupon-promo-code">
+          <Form.Label className="font-semibold text-grey-600">
+            <h2>Promo Code</h2>
+          </Form.Label>
+        </Localized>
 
-      <form action={formAction} data-testid="coupon-form">
-        <div className="flex gap-4 justify-between items-center">
+        <div className="mt-4 flex gap-4 justify-between items-center">
           <Localized attrs={{ placeholder: true }} id="next-coupon-enter-code">
```

Comment:
If you move this `<Localized>` inside `<Form.Control asChild>`, so that it wraps the `<input>` it should solve the issue with the placeholder translation.